File: POS System - Retaurant - 2022/POS System/StockPosition.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace POS_System
{

    public partial class StockPosition
    {
        private bool drag;
        private int mousex, mousey, Pos;

        public StockPosition()
        {
            InitializeComponent();
        }

        private void frmStockPosition_MouseDown(object sender, MouseEventArgs e)
        {
            drag = true;
            mousex = Cursor.Position.X - Left;
            mousey = Cursor.Position.Y - Top;
        }

        private void frmStockPosition_MouseMove(object sender, MouseEventArgs e)
        {
            if (drag)
            {
                Top = Cursor.Position.Y - mousey;
                Left = Cursor.Position.X - mousex;
            }
        }

        private void frmStockPosition_MouseUp(object sender, MouseEventArgs e)
        {
            drag = false;
        }

        private void cmdClose_Click(object sender, EventArgs e)
        {
            Close();
            My.MyProject.Forms.Stock.WindowState = FormWindowState.Normal;
        }

        private void cmdMin_Click(object sender, EventArgs e)
        {
            WindowState = FormWindowState.Minimized;
        }

        private void frmStockPosition_Load(object sender, EventArgs e)
        {
            FormBorderStyle = FormBorderStyle.None;
            cmdClose.Left = Width - (cmdClose.Width + 5);
            cmdMin.Left = Width - (cmdMin.Width + cmdClose.Width + 5);
            mdlGeneral.GetConnectionString();
            try
            {
                // Retrieve Records for datagrid view from product table
                ProductDataSet.Clear();
                mdlGeneral.ProductAdapter.Dispose();
                string ProductQuery = "SELECT * FROM PRODUCTS ORDER BY PROD_CD";
                mdlGeneral.ExecuteAdapterQuery(mdlGeneral.ProductAdapter, ProductQuery, mdlGeneral.Connection);
                mdlGeneral.ProductAdapter.Fill(ProductDataSet, "PRODUCTS");
                var reader2 = mdlGeneral.ProductAdapter.SelectCommand.ExecuteReader();
                if (reader2.HasRows)
                {
                    ProductBindingSource.DataSource = ProductDataSet;
                    ProductBindingSource.DataMember = "PRODUCTS";
                    ProductDataGridView.Refresh();
                }
            }
            catch (Exception EX)
            {
                Interaction.MsgBox(EX.Message, MsgBoxStyle.Information);
            }
            BindingNavigatorSummary_Click(sender, e);

        }

        private void BindingNavigatorFind_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrEmpty(BindingNavigatorTextBox.Text))
            {
                Interaction.MsgBox("Please give valid product code.", (MsgBoxStyle)((int)MsgBoxStyle.Information + (int)MsgBoxStyle.OkOnly), My.MyProject.Application.Info.ProductName);
                return;
            }
            string strCode;
            strCode = BindingNavigatorTextBox.Text.Trim();
            if (Strings.Len(strCode) == 0)
                return;
            Pos = ProductBindingSource.Find("PROD_CD", strCode.Trim());
            if (!(Pos == -1))
            {
                ProductBindingSource.Position = Pos;
            }
        }

        private void BindingNavigatorTextBox_Click(object sender, EventArgs e)
        {
            BindingNavigatorFind_Click(sender, e);
        }

        private void BindingNavigatorTextBox_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                BindingNavigatorFind_Click(sender, e);
            }
        }

        private void BindingNavigatorSummary_Click(object sender, EventArgs e)
        {

            if (DataGridView1.Rows.Count > 0)
                DataGridView1.Rows.Clear();

            string Summary_Query = "SELECT SUM(PROD_ONHAND) AS RAW_STOCK, " + "SUM(PROD_PPRICE * PROD_ONHAND) AS TOTAL_AMNT, " + "SUM(PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND) AS SCHEME_STOCK, " + "SUM((PROD_SCH1_PRICE * PROD_SCH1_ONHAND) + (PROD_SCH2_PRICE * PROD_SCH2_ONHAND) + (PROD_SCH3_PRICE * PROD_SCH3_ONHAND)) AS SCHEME_AMOUNT, " + "SUM(PROD_PROFIT * (PROD_ONHAND + PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND)) AS PROFIT " + "FROM PRODUCTS WHERE PROD_ONHAND >= 0";





            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, Summary_Query, mdlGeneral.Connection);
            var SummaryReader = mdlGeneral.SqlDataAdapter.SelectCommand.ExecuteReader();
            // DataGridView1.ColumnCount = 9
            if (SummaryReader.HasRows)
            {
                while (SummaryReader.Read())
                {
                    if (Conversion.Val(SummaryReader["TOTAL_AMNT"].ToString()) > 0d)
                    {
                        int RowIndex = DataGridView1.Rows.Add();
                        var NewRow = DataGridView1.Rows[RowIndex];
                        NewRow.Cells[0].Value = "Stock with Positive Balance";
                        NewRow.Cells[1].Value = SummaryReader["RAW_STOCK"].ToString();
                        NewRow.Cells[2].Value = SummaryReader["TOTAL_AMNT"].ToString();
                        NewRow.Cells[3].Value = SummaryReader["SCHEME_STOCK"].ToString();
                        NewRow.Cells[4].Value = SummaryReader["SCHEME_AMOUNT"].ToString();
                        NewRow.Cells[5].Value = SummaryReader["PROFIT"].ToString();
                    }
                }
                if (SummaryReader is null)
                    SummaryReader.Close();
            }

            string Summary_Query_Neg = "SELECT SUM(PROD_ONHAND) AS RAW_STOCK, " + "SUM(PROD_PPRICE * PROD_ONHAND) AS TOTAL_AMNT, " + "SUM(PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND) AS SCHEME_STOCK, " + "SUM((PROD_SCH1_PRICE * PROD_SCH1_ONHAND) + (PROD_SCH2_PRICE * PROD_SCH2_ONHAND) + (PROD_SCH3_PRICE * PROD_SCH3_ONHAND)) AS SCHEME_AMOUNT, " + "SUM(PROD_PROFIT * (PROD_ONHAND + PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND)) AS PROFIT " + "FROM PRODUCTS WHERE PROD_ONHAND < 0";





            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, Summary_Query_Neg, mdlGeneral.Connection);
            var SummaryReaderNeg = mdlGeneral.SqlDataAdapter.SelectCommand.ExecuteReader();
            // DataGridView1.ColumnCount = 9
            if (SummaryReaderNeg.HasRows)
            {
                while (SummaryReaderNeg.Read())
                {
                    if (Conversion.Val(SummaryReaderNeg["TOTAL_AMNT"].ToString()) < 0d)
                    {
                        int RowIndex = DataGridView1.Rows.Add();
                        var NewRow = DataGridView1.Rows[RowIndex];
                        NewRow.Cells[0].Value = "Stock with Negative Balance";
                        NewRow.Cells[1].Value = SummaryReader["RAW_STOCK"].ToString();
                        NewRow.Cells[2].Value = SummaryReader["TOTAL_AMNT"].ToString();
                        NewRow.Cells[3].Value = SummaryReader["SCHEME_STOCK"].ToString();
                        NewRow.Cells[4].Value = SummaryReader["SCHEME_AMOUNT"].ToString();
                        NewRow.Cells[5].Value = SummaryReader["PROFIT"].ToString();
                    }
                }
                if (SummaryReaderNeg is null)
                    SummaryReaderNeg.Close();
            }

        }

        private void BindingNavigatorSummaryGroup_Click(object sender, EventArgs e)
        {
            if (DataGridView1.Rows.Count > 0)
                DataGridView1.Rows.Clear();

            string Summary_Query = "SELECT PROD_CATG, " + "SUM(PROD_ONHAND) AS RAW_STOCK, " + "SUM(PROD_PPRICE * PROD_ONHAND) AS TOTAL_AMNT, " + "SUM(PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND) AS SCHEME_STOCK, " + "SUM((PROD_SCH1_PRICE * PROD_SCH1_ONHAND) + (PROD_SCH2_PRICE * PROD_SCH2_ONHAND) + (PROD_SCH3_PRICE * PROD_SCH3_ONHAND)) AS SCHEME_AMOUNT, " + "SUM(PROD_PROFIT * (PROD_ONHAND + PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND)) AS PROFIT " + "FROM PRODUCTS WHERE PROD_ONHAND >= 0 " + "GROUP BY PROD_CATG";







            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, Summary_Query, mdlGeneral.Connection);
            var SummaryReader = mdlGeneral.SqlDataAdapter.SelectCommand.ExecuteReader();
            // DataGridView1.ColumnCount = 9
            if (SummaryReader.HasRows)
            {
                while (SummaryReader.Read())
                {
                    if (Conversion.Val(SummaryReader["TOTAL_AMNT"].ToString()) > 0d)
                    {
                        int RowIndex = DataGridView1.Rows.Add();
                        var NewRow = DataGridView1.Rows[RowIndex];
                        NewRow.Cells[0].Value = SummaryReader["PROD_CATG"].ToString().Trim() + " with +ve Balance";
                        NewRow.Cells[1].Value = SummaryReader["RAW_STOCK"].ToString();
                        NewRow.Cells[2].Value = SummaryReader["TOTAL_AMNT"].ToString();
                        NewRow.Cells[3].Value = SummaryReader["SCHEME_STOCK"].ToString();
                        NewRow.Cells[4].Value = SummaryReader["SCHEME_AMOUNT"].ToString();
                        NewRow.Cells[5].Value = SummaryReader["PROFIT"].ToString();
                    }
                }
                if (SummaryReader is null)
                    SummaryReader.Close();
            }

            string Summary_Query_Neg = "SELECT PROD_CATG, " + "SUM(PROD_ONHAND) AS RAW_STOCK, " + "SUM(PROD_PPRICE * PROD_ONHAND) AS TOTAL_AMNT, " + "SUM(PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND) AS SCHEME_STOCK, " + "SUM((PROD_SCH1_PRICE * PROD_SCH1_ONHAND) + (PROD_SCH2_PRICE * PROD_SCH2_ONHAND) + (PROD_SCH3_PRICE * PROD_SCH3_ONHAND)) AS SCHEME_AMOUNT, " + "SUM(PROD_PROFIT * (PROD_ONHAND + PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND)) AS PROFIT " + "FROM PRODUCTS WHERE PROD_ONHAND < 0 " + "GROUP BY PROD_CATG";







            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, Summary_Query_Neg, mdlGeneral.Connection);
            var SummaryReaderNeg = mdlGeneral.SqlDataAdapter.SelectCommand.ExecuteReader();
            // DataGridView1.ColumnCount = 9
            if (SummaryReaderNeg.HasRows)
            {
                while (SummaryReaderNeg.Read())
                {
                    if (Conversion.Val(SummaryReaderNeg["TOTAL_AMNT"].ToString()) < 0d)
                    {
                        int RowIndex = DataGridView1.Rows.Add();
                        var NewRow = DataGridView1.Rows[RowIndex];
                        NewRow.Cells[0].Value = SummaryReaderNeg["PROD_CATG"].ToString().Trim() + " with -ve Balance";
                        NewRow.Cells[1].Value = SummaryReader["RAW_STOCK"].ToString();
                        NewRow.Cells[2].Value = SummaryReader["TOTAL_AMNT"].ToString();
                        NewRow.Cells[3].Value = SummaryReader["SCHEME_STOCK"].ToString();
                        NewRow.Cells[4].Value = SummaryReader["SCHEME_AMOUNT"].ToString();
                        NewRow.Cells[5].Value = SummaryReader["PROFIT"].ToString();
                    }
                }
                if (SummaryReaderNeg is null)
                    SummaryReaderNeg.Close();
            }
        }

        private void BindingNavigatorTextBox_KeyUp(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.F2)
            {
                My.MyProject.Forms.ProductCode.ShowDialog(this);
                if (Strings.Len(mdlGeneral.gblItemCode) > 0)
                    BindingNavigatorTextBox.Text = mdlGeneral.gblItemCode.ToUpper();
                else
                    return;
            }
        }
    }
}

File: POS System/POS System/StockPosition.vb
Imports System.Data.Sql
Imports System.Data.SqlClient

Public Class StockPosition
    Dim drag As Boolean
    Dim mousex, mousey, Pos As Integer

    Private Sub frmStockPosition_MouseDown(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles Me.MouseDown
        drag = True
        mousex = Windows.Forms.Cursor.Position.X - Me.Left
        mousey = Windows.Forms.Cursor.Position.Y - Me.Top
    End Sub

    Private Sub frmStockPosition_MouseMove(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles Me.MouseMove
        If drag Then
            Me.Top = Windows.Forms.Cursor.Position.Y - mousey
            Me.Left = Windows.Forms.Cursor.Position.X - mousex
        End If
    End Sub

    Private Sub frmStockPosition_MouseUp(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles Me.MouseUp
        drag = False
    End Sub

    Private Sub cmdClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdClose.Click
        Me.Close()
        Stock.WindowState = FormWindowState.Normal
    End Sub

    Private Sub cmdMin_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdMin.Click
        Me.WindowState = FormWindowState.Minimized
    End Sub

    Private Sub frmStockPosition_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.FormBorderStyle = Windows.Forms.FormBorderStyle.None
        Me.cmdClose.Left = Me.Width - (cmdClose.Width + 5)
        Me.cmdMin.Left = Me.Width - (cmdMin.Width + cmdClose.Width + 5)
        Call GetConnectionString()
        Try
            'Retrieve Records for datagrid view from product table
            Me.ProductDataSet.Clear()
            ProductAdapter.Dispose()
            Dim ProductQuery As String = "SELECT * FROM PRODUCTS ORDER BY PROD_CD"
            ExecuteAdapterQuery(ProductAdapter, ProductQuery, Connection)
            ProductAdapter.Fill(Me.ProductDataSet, "PRODUCTS")
            Dim reader2 As SqlDataReader = ProductAdapter.SelectCommand.ExecuteReader
            If reader2.HasRows Then
                ProductBindingSource.DataSource = Me.ProductDataSet
                ProductBindingSource.DataMember = "PRODUCTS"
                ProductDataGridView.Refresh()
            End If
        Catch EX As Exception
            MsgBox(EX.Message, MsgBoxStyle.Information)
        End Try
        Call BindingNavigatorSummary_Click(sender, e)

    End Sub

    Private Sub BindingNavigatorFind_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorFind.Click
        If BindingNavigatorTextBox.Text = "" Then MsgBox("Please give valid product code.", MsgBoxStyle.Information + MsgBoxStyle.OkOnly, My.Application.Info.ProductName) : Exit Sub
        Dim strCode As String
        strCode = BindingNavigatorTextBox.Text.Trim
        If Len(strCode) = 0 Then Exit Sub
        Pos = Me.ProductBindingSource.Find("PROD_CD", strCode.Trim)
        If Not Pos = -1 Then
            Me.ProductBindingSource.Position = Pos
        End If
    End Sub

    Private Sub BindingNavigatorTextBox_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorTextBox.Click
        Call BindingNavigatorFind_Click(sender, e)
    End Sub

    Private Sub BindingNavigatorTextBox_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles BindingNavigatorTextBox.KeyDown
        If e.KeyCode = Keys.Enter Then
            Call BindingNavigatorFind_Click(sender, e)
        End If
    End Sub

    Private Sub BindingNavigatorSummary_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorSummary.Click

        If DataGridView1.Rows.Count > 0 Then DataGridView1.Rows.Clear()

        Dim Summary_Query As String = "SELECT SUM(PROD_ONHAND) AS RAW_STOCK, " & _
                                      "SUM(PROD_PPRICE * PROD_ONHAND) AS TOTAL_AMNT, " & _
                                      "SUM(PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND) AS SCHEME_STOCK, " & _
                                      "SUM((PROD_SCH1_PRICE * PROD_SCH1_ONHAND) + (PROD_SCH2_PRICE * PROD_SCH2_ONHAND) + (PROD_SCH3_PRICE * PROD_SCH3_ONHAND)) AS SCHEME_AMOUNT, " & _
                                      "SUM(PROD_PROFIT * (PROD_ONHAND + PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND)) AS PROFIT " & _
                                      "FROM PRODUCTS WHERE PROD_ONHAND >= 0"

        ExecuteAdapterQuery(SqlDataAdapter, Summary_Query, Connection)
        Dim SummaryReader As SqlDataReader = SqlDataAdapter.SelectCommand.ExecuteReader
        'DataGridView1.ColumnCount = 9
        If SummaryReader.HasRows Then
            While SummaryReader.Read
                If Val(SummaryReader.Item("TOTAL_AMNT").ToString) > 0 Then
                    Dim RowIndex As Integer = DataGridView1.Rows.Add()
                    Dim NewRow As DataGridViewRow = DataGridView1.Rows(RowIndex)
                    NewRow.Cells(0).Value = "Stock with Positive Balance"
                    NewRow.Cells(1).Value = SummaryReader.Item("RAW_STOCK").ToString
                    NewRow.Cells(2).Value = SummaryReader.Item("TOTAL_AMNT").ToString
                    NewRow.Cells(3).Value = SummaryReader.Item("SCHEME_STOCK").ToString
                    NewRow.Cells(4).Value = SummaryReader.Item("SCHEME_AMOUNT").ToString
                    NewRow.Cells(5).Value = SummaryReader.Item("PROFIT").ToString
                End If
            End While
            If SummaryReader Is Nothing Then SummaryReader.Close()
        End If

        Dim Summary_Query_Neg As String = "SELECT SUM(PROD_ONHAND) AS RAW_STOCK, " & _
                                          "SUM(PROD_PPRICE * PROD_ONHAND) AS TOTAL_AMNT, " & _
                                          "SUM(PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND) AS SCHEME_STOCK, " & _
                                          "SUM((PROD_SCH1_PRICE * PROD_SCH1_ONHAND) + (PROD_SCH2_PRICE * PROD_SCH2_ONHAND) + (PROD_SCH3_PRICE * PROD_SCH3_ONHAND)) AS SCHEME_AMOUNT, " & _
                                          "SUM(PROD_PROFIT * (PROD_ONHAND + PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND)) AS PROFIT " & _
                                          "FROM PRODUCTS WHERE PROD_ONHAND < 0"

        ExecuteAdapterQuery(SqlDataAdapter, Summary_Query_Neg, Connection)
        Dim SummaryReaderNeg As SqlDataReader = SqlDataAdapter.SelectCommand.ExecuteReader
        'DataGridView1.ColumnCount = 9
        If SummaryReaderNeg.HasRows Then
            While SummaryReaderNeg.Read
                If Val(SummaryReaderNeg.Item("TOTAL_AMNT").ToString) < 0 Then
                    Dim RowIndex As Integer = DataGridView1.Rows.Add()
                    Dim NewRow As DataGridViewRow = DataGridView1.Rows(RowIndex)
                    NewRow.Cells(0).Value = "Stock with Negative Balance"
                    NewRow.Cells(1).Value = SummaryReader.Item("RAW_STOCK").ToString
                    NewRow.Cells(2).Value = SummaryReader.Item("TOTAL_AMNT").ToString
                    NewRow.Cells(3).Value = SummaryReader.Item("SCHEME_STOCK").ToString
                    NewRow.Cells(4).Value = SummaryReader.Item("SCHEME_AMOUNT").ToString
                    NewRow.Cells(5).Value = SummaryReader.Item("PROFIT").ToString
                End If
            End While
            If SummaryReaderNeg Is Nothing Then SummaryReaderNeg.Close()
        End If

    End Sub

    Private Sub BindingNavigatorSummaryGroup_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorSummaryGroup.Click
        If DataGridView1.Rows.Count > 0 Then DataGridView1.Rows.Clear()

        Dim Summary_Query As String = "SELECT PROD_CATG, " & _
                                      "SUM(PROD_ONHAND) AS RAW_STOCK, " & _
                                      "SUM(PROD_PPRICE * PROD_ONHAND) AS TOTAL_AMNT, " & _
                                      "SUM(PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND) AS SCHEME_STOCK, " & _
                                      "SUM((PROD_SCH1_PRICE * PROD_SCH1_ONHAND) + (PROD_SCH2_PRICE * PROD_SCH2_ONHAND) + (PROD_SCH3_PRICE * PROD_SCH3_ONHAND)) AS SCHEME_AMOUNT, " & _
                                      "SUM(PROD_PROFIT * (PROD_ONHAND + PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND)) AS PROFIT " & _
                                      "FROM PRODUCTS WHERE PROD_ONHAND >= 0 " & _
                                      "GROUP BY PROD_CATG"

        ExecuteAdapterQuery(SqlDataAdapter, Summary_Query, Connection)
        Dim SummaryReader As SqlDataReader = SqlDataAdapter.SelectCommand.ExecuteReader
        'DataGridView1.ColumnCount = 9
        If SummaryReader.HasRows Then
            While SummaryReader.Read
                If Val(SummaryReader.Item("TOTAL_AMNT").ToString) > 0 Then
                    Dim RowIndex As Integer = DataGridView1.Rows.Add()
                    Dim NewRow As DataGridViewRow = DataGridView1.Rows(RowIndex)
                    NewRow.Cells(0).Value = SummaryReader.Item("PROD_CATG").ToString.Trim & " with +ve Balance"
                    NewRow.Cells(1).Value = SummaryReader.Item("RAW_STOCK").ToString
                    NewRow.Cells(2).Value = SummaryReader.Item("TOTAL_AMNT").ToString
                    NewRow.Cells(3).Value = SummaryReader.Item("SCHEME_STOCK").ToString
                    NewRow.Cells(4).Value = SummaryReader.Item("SCHEME_AMOUNT").ToString
                    NewRow.Cells(5).Value = SummaryReader.Item("PROFIT").ToString
                End If
            End While
            If SummaryReader Is Nothing Then SummaryReader.Close()
        End If

        Dim Summary_Query_Neg As String = "SELECT PROD_CATG, " & _
                                          "SUM(PROD_ONHAND) AS RAW_STOCK, " & _
                                          "SUM(PROD_PPRICE * PROD_ONHAND) AS TOTAL_AMNT, " & _
                                          "SUM(PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND) AS SCHEME_STOCK, " & _
                                          "SUM((PROD_SCH1_PRICE * PROD_SCH1_ONHAND) + (PROD_SCH2_PRICE * PROD_SCH2_ONHAND) + (PROD_SCH3_PRICE * PROD_SCH3_ONHAND)) AS SCHEME_AMOUNT, " & _
                                          "SUM(PROD_PROFIT * (PROD_ONHAND + PROD_SCH1_ONHAND + PROD_SCH2_ONHAND + PROD_SCH3_ONHAND)) AS PROFIT " & _
                                          "FROM PRODUCTS WHERE PROD_ONHAND < 0 " & _
                                          "GROUP BY PROD_CATG"

        ExecuteAdapterQuery(SqlDataAdapter, Summary_Query_Neg, Connection)
        Dim SummaryReaderNeg As SqlDataReader = SqlDataAdapter.SelectCommand.ExecuteReader
        'DataGridView1.ColumnCount = 9
        If SummaryReaderNeg.HasRows Then
            While SummaryReaderNeg.Read
                If Val(SummaryReaderNeg.Item("TOTAL_AMNT").ToString) < 0 Then
                    Dim RowIndex As Integer = DataGridView1.Rows.Add()
                    Dim NewRow As DataGridViewRow = DataGridView1.Rows(RowIndex)
                    NewRow.Cells(0).Value = SummaryReaderNeg.Item("PROD_CATG").ToString.Trim & " with -ve Balance"
                    NewRow.Cells(1).Value = SummaryReader.Item("RAW_STOCK").ToString
                    NewRow.Cells(2).Value = SummaryReader.Item("TOTAL_AMNT").ToString
                    NewRow.Cells(3).Value = SummaryReader.Item("SCHEME_STOCK").ToString
                    NewRow.Cells(4).Value = SummaryReader.Item("SCHEME_AMOUNT").ToString
                    NewRow.Cells(5).Value = SummaryReader.Item("PROFIT").ToString
                End If
            End While
            If SummaryReaderNeg Is Nothing Then SummaryReaderNeg.Close()
        End If
    End Sub

    Private Sub BindingNavigatorTextBox_KeyUp(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles BindingNavigatorTextBox.KeyUp
        If e.KeyCode = Keys.F2 Then
            ProductCode.ShowDialog(Me)
            If Len(gblItemCode) > 0 Then BindingNavigatorTextBox.Text = gblItemCode.ToUpper Else Exit Sub
        End If
    End Sub
End Class